mdata/aggregator.go

```diff
@@ -99,6 +100,38 @@ func (agg *Aggregator) flush() {
 	agg.agg.Reset()
 }
 
+// Foresee duplicates the underlying Aggregation,
+// consumes points in the future, and return the newly aggregated points.
+// Foresee won't actually add those points to AggMetric.
+func (agg *Aggregator) Foresee(consolidator consolidation.Consolidator, from, to uint32, futurePoints []schema.Point) []schema.Point {
+	aggregationPreview := *(agg.agg)
```

Comment:
wow this is clever. so we make a copy of the aggregator, scoped to only this request, to temporarily add the ROB's points to, so we can compute the full aggregation. nice... i could not come up with this.